File: powerset.go
package main
import "fmt"
func powerSet(str string,index int,curr string){
	var n=len(str)
	if index==n{
		fmt.Printf("{%s}, ",curr)
		return
	}
	powerSet(str,index+1,curr+string(str[index]))
	powerSet(str,index+1,curr)
}
func main(){
	var str string
	fmt.Printf("Enter a string: ")
	fmt.Scanf("%s",&str)
	fmt.Printf("Sets of Inputted String are: ")
	powerSet(str,0,"")
	fmt.Printf("\n")
}

File: powerSet.js
function powerSet(array) {
  var result = [];
  result.push([]);
  //sets in power set is pow(2,n) which we can get by (1<<array.length)
  for (var i = 1; i < (1 << array.length); i++) {
    var subset = [];
    
    for (var j = 0; j < array.length; j++)
     if (i & (1 << j))
        subset.push(array[j]);
     //push the sets in the subset array
      
    result.push(subset);
  }

  return result;
}
//test
let inputArray=[];
let size=prompt("enter size of array");
for(let i=0;i<size;i++)
  {
    inputArray[i]=prompt("enter sorted array");
    inputArray[i]=parseInt(inputArray[i]);
  }
console.log(powerSet(inputArray));